### PR TITLE
UX: Site setting descriptions initial pass

### PR DESCRIPTION
**Pull request by martin-brennan**
This commit changes the wording of the 50 site settings that
previously had the shortest descriptions (e.g. City for Disputes
was described as City for Disputes...) using AI-generated and then
human curated descriptions based on the Forum Helper persona
on Discourse Meta.

In future we may want to do more of these, this is only a first pass.

### CODE REVIEW

File: config/locales/server.en.yml [hunk 1]
```diff
@@ -1719,7 +1719,7 @@ en:
     blocked_crawler_user_agents: "Unique case insensitive word in the user agent string identifying web crawlers that should not be allowed to access the site. Does not apply if allowlist is defined."
     slow_down_crawler_user_agents: 'User agents of web crawlers that should be rate limited as configured in the "slow down crawler rate" setting. Each value must be at least 3 characters long.'
     slow_down_crawler_rate: "If slow_down_crawler_user_agents is specified this rate will apply to all the crawlers (number of seconds delay between requests)"
-    content_security_policy: "Enable Content-Security-Policy (CSP)"
+    content_security_policy: "Activates the Content-Security-Policy (CSP), an additional layer of security that helps to prevent certain types of attacks, including Cross Site Scripting (XSS) and data injection. When enabled, the server provides the security policy in the form of a 'Content-Security-Policy' HTTP header."
```
Comment (pmusaraj) [inline config/locales/server.en.yml:1722]: I would keep "Enable Content-Security-Policy (CSP)`, we usually don't use the "Activate" wording for enabling site settings. Also the phrase "When enabled, the server provides the security policy in the form of a 'Content-Security-Policy' HTTP header" is redundant, admins shouldn't care how the CSP is delivered. (And I'm pretty sure it can _only_ be a header.) 

```suggestion
    content_security_policy: "Enable Content-Security-Policy (CSP). CSP is an additional layer of security that helps to prevent certain types of attacks, including Cross Site Scripting (XSS) and data injection."
```

File: config/locales/server.en.yml [hunk 2]
```diff
@@ -1838,15 +1838,15 @@ en:
 
     enable_discord_logins: "Allow users to authenticate using Discord?"
     discord_client_id: 'Discord Client ID (need one? visit <a href="https://discordapp.com/developers/applications/me">the Discord developer portal</a>)'
-    discord_secret: "Discord Secret Key"
+    discord_secret: "Discord Client Secret Key Used for authenticating and enabling Discord related features on the site, such as Discord logins. This secret key corresponds to the Discord application created for the website, and is necessary for securely communicating with the Discord API. Misuse may lead to unauthorized access so handle it carefully."
```
Comment (pmusaraj) [inline config/locales/server.en.yml:1841]:
```
Misuse may lead to unauthorized access so handle it carefully.
```

I'm not sure what this means... feels unnecessary.

File: config/locales/server.en.yml [hunk 3]
```diff
@@ -2307,7 +2307,7 @@ en:
     app_association_android: "Contents of <a href='%{base_path}/.well-known/assetlinks.json'>.well-known/assetlinks.json</a> endpoint, used for Google's Digital Asset Links API."
     app_association_ios: "Contents of <a href='%{base_path}/apple-app-site-association'>apple-app-site-association</a> endpoint, used to create Universal Links between this site and iOS apps."
 
-    share_anonymized_statistics: "Share anonymized usage statistics."
+    share_anonymized_statistics: "Enable sharing of anonymized usage statistics. When this setting is enabled, data concerning the usage of the site is collected and shared in anonymized form, ensuring no personal information is disclosed."
```
Comment (pmusaraj) [inline config/locales/server.en.yml:2310]: Should this setting say who the anonymized stats are shared with (either Discourse or CDCK Inc.)?

File: config/locales/server.en.yml [hunk 4]
```diff
@@ -2348,9 +2348,9 @@ en:
     secure_uploads_max_email_embed_image_size_kb: "The size cutoff for secure images that will be embedded in emails if the 'secure uploads allow embed in emails' setting is enabled. Without that setting enabled, this setting has no effect."
     slug_generation_method: "Choose a slug generation method. 'encoded' will generate percent encoding string. 'none' will disable slug at all."
 
-    enable_emoji: "Enable emoji"
+    enable_emoji: "Toggle the display and use of emojis in your Discourse instance. When enabled, users have the ability to insert emojis into their posts and other text-related fields, enhancing expressive communication. This also enables the emoji autocomplete function in the text editor. If disabled, emojis will not be rendered and users will not be able to access or use them."
```
Comment (pmusaraj) [inline config/locales/server.en.yml:2351]: This is so very wordy... I'm not sure it's better than the old, concise "Enable emoji" label.

File: config/locales/server.en.yml (hunk 1 above)
Comment (martin-brennan) [reply config/locales/server.en.yml:1722]: Sounds good :+1:

File: config/locales/server.en.yml (hunk 3 above)
Comment (martin-brennan) [reply config/locales/server.en.yml:2310]: Good point, will update to say CDCK Inc :+1:

File: config/locales/server.en.yml (hunk 2 above)
Comment (martin-brennan) [reply config/locales/server.en.yml:1841]: Yep will remove

File: config/locales/server.en.yml (hunk 4 above)
Comment (martin-brennan) [reply config/locales/server.en.yml:2351]: We are trying to get rid of the redundant "enable_x" -> "Enable X" descriptions. It is a bit wordy though, will cut it down :+1:

**Comment by martin-brennan** [comment]
@pmusaraj Updated based on suggestions :)

**Review by pmusaraj — APPROVED** [review]
